compiler/src/iree/compiler/Codegen/Dialect/VectorExt/IR/VectorExtOps.td

```diff
@@ -40,21 +40,31 @@ def IREEVectorExt_ToLayoutOp : IREEVectorExt_PureOp<"to_layout", [
   let arguments = (ins
     AnyShaped:$input,
     VectorLayoutInterface:$layout,
-    DefaultValuedAttr<UnitAttr, "false">:$shared_memory_conversion
+    DefaultValuedAttr<UnitAttr, "false">:$shared_memory_conversion,
+    // TODO: Solve cmake IREEGPU and VectorExt cyclic dependency to
+    // change mma_Kind type to be of MMAInterfaceAttr.
+    OptionalAttr<AnyAttr>:$mma_kind
```

Comment:
I would recommend swapping these two attributes. default valued parameters should be at end I think. It makes the constructors easier.

Reply:
hmmm, shouldn't optionalAttr be at the end? haha it's interesting because one is OptionalAttr, another is DefaultValuedAttr

Reply:
It's weird yeah, but i think if you put default valued attr, it creates a constructor with a default value, while it doesn't do that for optional attr. So it's better to put at the end.

Reply:
actually, reading the mlir reference, default valued attrs should also have a default argument which can be passed to the constructor.

Reply:
It's fine, I think i misunderstood. Ignore my earlier comment and keep it this way.

Reply:
cool cool, landed :)